RIOT-OS/RIOT
# makefiles: add FEATURES_OPTIONAL_ANY

### Contribution description

From looking at  #15836 and #15817, it's clear that some dependencies are better modeled in Kconfig with choices. This PR looks to give a similar option to model this behavior in `make` while the migration is ongoing.

So similar to #13349:

> - `FEATURES_OPTIONAL_ALT` are nice to have `FEATURES` that fulfill a similar
>   function. Alternatives are separated by a pipe (`|`) in order of preference,
>   e.g.: `FEATURES_OPTIONAL_ALT += puf_sram|periph_hwrng` if both are provided
>   then only `puf_sram` will be used.


### Testing procedure

#12166 Shows a test case for this (I want to rebase that one on top of  #15836 and this one)

The save_all_dependencies script should have the same output as before, as this PR does not make use of the feature.

### Issues/PRs references

Taken from #12166
Related to  #15836
